config/crd/bases/config.nri_nriplugindeployments.yaml

```diff
@@ -69,8 +69,11 @@ spec:
                     nri:
                       type: object
                       properties:
-                        patchRuntimeConfig:
-                          type: boolean
+                        runtime:
+                          type: object
+                          properties:
+                            patchConfig:
+                              type: boolean
```

Comment:
I bet we need to update the CRD here to include timeouts if we want consumers of the operator to be able to tune the timeouts. 
And if we do update the CRD here, we should also update the [README](https://github.com/containers/nri-plugins/blob/main/deployment/operator/README.md).

Reply:
It's updated in the next commit...

Reply:
Sure, I will run the operator and let you know.

Reply:
@klihub here is what I got
```
fmuyassarov@fmuyassa-MOBL1:/tmp/nri-plugins$ make -C deployment/operator deploy IMAGE=feruzjon/nri-plugins-operator:b4fde25552e46840d9c88da7f0fed989e6bb33dc
make: Entering directory '/tmp/nri-plugins/deployment/operator'
mkdir -p config/crd/bases
cp ../../config/crd/bases/* config/crd/bases
cp ../../sample-configs/balloons-config.yaml config/samples
cp ../../sample-configs/template-config.yaml config/samples
cp ../../sample-configs/topologyaware-config.yaml config/samples
cp ../../sample-configs/noderesourcetopology.yaml config/samples
cd config/manager && /usr/local/bin/kustomize edit set image controller=feruzjon/nri-plugins-operator:b4fde25552e46840d9c88da7f0fed989e6bb33dc
/usr/local/bin/kustomize build config/default | kubectl apply -f -
Error: accumulating resources: accumulation err='accumulating resources from '../crd': '/tmp/nri-plugins/deployment/operator/config/crd' must resolve to a file': recursed accumulation of path '/tmp/nri-plugins/deployment/operator/config/crd': accumulating resources: accumulating resources from 'bases/config.nri_nriplugindeployments.yaml': MalformedYAMLError: yaml: line 82: mapping values are not allowed in this context in File: bases/config.nri_nriplugindeployments.yaml
error: no objects passed to apply
make: *** [Makefile:186: deploy] Error 1
make: Leaving directory '/tmp/nri-plugins/deployment/operator'
```

Reply:
(I think I) Fixed (it).

Reply:
@klihub one more issue to be fixed but we are close
```diff
diff --git a/config/crd/bases/config.nri_nriplugindeployments.yaml b/config/crd/bases/config.nri_nriplugindeployments.yaml
index 38f4134..4d0ff56 100644
--- a/config/crd/bases/config.nri_nriplugindeployments.yaml
+++ b/config/crd/bases/config.nri_nriplugindeployments.yaml
@@ -74,19 +74,18 @@ spec:
                           properties:
                             patchConfig:
                               type: boolean
-                          config:
-                            type: object
-                            properties:
+                            config:
+                              properties:
+                                pluginRegistrationTimeout:
+                                  type: string
+                                  pattern: "^(([5-9])|([1-2][0-9])|(30))s$"
+                                pluginRequestTimeout:
+                                  type: string
+                                  pattern: "^(([2-9])|([1-2][0-9])|(30))s$"
+                              type: object
                               required:
-                                - pluginRegistrationTimeout
-                                - pluginRequestTimeout
-                              pluginRegistrationTimeout:
-                                type: string
-                                pattern: "^(([5-9])|([1-2][0-9])|(30))s$"
-                              pluginRequestTimeout:
-                                type: string
-                                pattern: "^(([2-9])|([1-2][0-9])|(30))s$"
-
+                              - pluginRegistrationTimeout
+                              - pluginRequestTimeout
                     image:
                       type: object
                       description: image defines Plugin DeamonSet image name and tag

```

Reply:
Argh, thanks for spotting that !